valhalla/thor/costmatrix.h

```diff
@@ -99,55 +97,29 @@ class CostMatrix {
    * @param  mode_costing          List of target/destination locations.
    * @param  mode                  Graph reader for accessing routing graph.
    * @param  max_matrix_distance   Maximum arc-length distance for current mode.
-   * @param  has_time              whether time-dependence was requested
-   * @param  invariant             whether invariant time-dependence was requested
-   * @param  shape_format          which shape_format, if any
    */
-  void SourceToTarget(Api& request,
+  bool SourceToTarget(Api& request,
                       baldr::GraphReader& graphreader,
                       const sif::mode_costing_t& mode_costing,
                       const sif::travel_mode_t mode,
-                      const float max_matrix_distance,
-                      const bool has_time = false,
-                      const bool invariant = false,
-                      const ShapeFormat& shape_format = no_shape);
+                      const float max_matrix_distance) override;
 
   /**
    * Clear the temporary information generated during time+distance
    * matrix construction.
    */
-  void clear();
+  void Clear() override;
 
   /**
-   * Sets the functor which will track the Dijkstra expansion.
-   *
-   * @param  expansion_callback  the functor to call back when the Dijkstra makes progress
-   *                             on a given edge
-   */
-  using expansion_callback_t = std::function<void(baldr::GraphReader&,
-                                                  const baldr::GraphId,
-                                                  const baldr::GraphId,
-                                                  const char*,
-                                                  const char*,
-                                                  float,
-                                                  uint32_t,
-                                                  float)>;
-  void set_track_expansion(const expansion_callback_t& expansion_callback) {
-    expansion_callback_ = expansion_callback;
-  }
-
-  /**
-   * Set a callback that will throw when the path computation should be aborted
-   * @param interrupt_callback  the function to periodically call to see if
-   *                            we should abort
+   * Get the algorithm's name
+   * @return the name of the algorithm
    */
-  void set_interrupt(const std::function<void()>* interrupt_callback) {
-    interrupt_ = interrupt_callback;
+  inline std::string_view name() override {
+    return "costmatrix";
```

Comment:
this should probably be a `constexpr`

Reply:
Actually we can use the proto_conversion stuff here too, so we only have one place where we hard-code those names